terraform/modules/iam/iam.tf

```diff
@@ -0,0 +1,111 @@
+      {
+        Effect = "Allow"
+        Action = [
+          "ecr:GetAuthorizationToken",
+          "ecr:BatchCheckLayerAvailability",
+          "ecr:GetDownloadUrlForLayer",
+          "ecr:BatchGetImage"
+        ]
+        Resource = [
+          "arn:aws:ecs:${var.region}:${var.aws_caller_identity}:cluster/${var.ecs_cluster_name}/*",
+          "arn:aws:ecs:${var.region}:${var.aws_caller_identity}:cluster/${var.ecs_cluster_name}",
+          "arn:aws:ecs:${var.region}:${var.aws_caller_identity}:service/${var.ecs_cluster_name}/ecr-viewer/*",
+          "arn:aws:ecs:${var.region}:${var.aws_caller_identity}:service/${var.ecs_cluster_name}/ecr-viewer",
+          "arn:aws:ecs:${var.region}:${var.aws_caller_identity}:service/${var.ecs_cluster_name}/orchestration/*",
+          "arn:aws:ecs:${var.region}:${var.aws_caller_identity}:service/${var.ecs_cluster_name}/orchestration",
+          "arn:aws:ecs:${var.region}:${var.aws_caller_identity}:service/${var.ecs_cluster_name}/fhir-converter/*",
+          "arn:aws:ecs:${var.region}:${var.aws_caller_identity}:service/${var.ecs_cluster_name}/fhir-converter",
+          "arn:aws:ecs:${var.region}:${var.aws_caller_identity}:service/${var.ecs_cluster_name}/ingestion/*",
+          "arn:aws:ecs:${var.region}:${var.aws_caller_identity}:service/${var.ecs_cluster_name}/ingestion",
+          "arn:aws:ecs:${var.region}:${var.aws_caller_identity}:service/${var.ecs_cluster_name}/validation/*",
+          "arn:aws:ecs:${var.region}:${var.aws_caller_identity}:service/${var.ecs_cluster_name}/validation",
+        ]
+      }
+    ],
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "logs:CreateLogGroup",
+          "logs:CreateLogStream"
+        ]
+        Resource = [
+          "*"
+        ]
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "ecs_task_execution_role_policy_secondary_attachment" {
+  role       = aws_iam_role.ecs_task_execution_role.name
+  policy_arn = aws_iam_policy.ecr_policy.arn
+}
+
+data "aws_iam_policy_document" "ecs_task_definition_execution_policy" {
+  statement {
+    actions = [
+      "ec2:DescribeNetworkInterfaces",
+      "ec2:CreateNetworkInterface",
+      "ec2:DeleteNetworkInterface",
+      "ec2:DescribeInstances",
+      "ec2:AttachNetworkInterface",
+      "servicediscovery:DiscoverInstances"
+    ]
+    resources = [
+      "arn:aws:eks:${var.region}:${var.aws_caller_identity}:cluster/phdi-playground-dev",
+      "arn:aws:eks:${var.region}:${var.aws_caller_identity}:cluster/phdi-playground-dev/*",
+      "arn:aws:ecs:${var.region}:${var.aws_caller_identity}:cluster/${var.ecs_cluster_name}/*",
+      "arn:aws:ecs:${var.region}:${var.aws_caller_identity}:cluster/${var.ecs_cluster_name}",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "ecs_cloudwatch_policy" {
+  statement {
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents"
+    ]
+    resources = [
```

Comment:
Do we need to add the `message-parser` and `trigger code reference` to this iam policy document?

Reply:
This is another case where it seems like we don't, logs are successfully being pushed to cloudwatch. I'll fiddle with this to understand why.

Reply:
This should be related to the latest answer here: https://github.com/CDCgov/dibbs-aws/pull/1#discussion_r1661506339